ccm_web/client/src/pages/BulkSectionCreate.tsx

```diff
@@ -0,0 +1,534 @@
+  dialog: {
+    textAlign: 'center',
+    marginBottom: 15,
+    paddingLeft: 10,
+    paddingRight: 10
+  },
+  table: {
+    paddingLeft: 10,
+    paddingRight: 10
+  },
+  dialogIcon: {
+    color: 'red'
+  }
+}))
+
+const useTopLevelErrorStyles = makeStyles((theme) => ({
+  dialog: {
+    textAlign: 'center',
+    maxWidth: '75%',
+    margin: 'auto',
+    marginBottom: 15,
+    paddingLeft: 10,
+    paddingRight: 10,
+    width: '75%',
+    '& ol': {
+      margin: 'auto',
+      width: '75%'
+    },
+    '& li': {
+      textAlign: 'left'
+    }
+  },
+  dialogIcon: {
+    color: 'red'
+  }
+}))
+
+const useAPIErrorStyles = makeStyles((theme) => ({
+  dialog: {
+    textAlign: 'center',
+    maxWidth: '75%',
+    margin: 'auto',
+    marginTop: 30,
+    marginBottom: 15,
+    paddingLeft: 10,
+    paddingRight: 10,
+    width: '75%',
+    '& ol': {
+      margin: 'auto',
+      width: '75%'
+    },
+    '& li': {
+      textAlign: 'left'
+    }
+  },
+  dialogIcon: {
+    color: 'red'
+  }
+}))
+
+enum BulkSectionCreatePageState {
+  LoadingExistingSectionNames,
+  LoadingExistingSectionNamesFailed,
+  Upload,
+  InvalidUpload,
+  Confirm,
+  Done
+}
+
+interface BulkSectionCreatePageStateData {
+  state: BulkSectionCreatePageState
+  rowInvalidations: SectionsRowInvalidation[]
+  schemaInvalidation: SectionsSchemaInvalidation[]
+}
+
+const isHeader = (text: string): boolean => {
+  return text.toUpperCase() === FILE_HEADER.toUpperCase()
+}
+
+const hasHeader = (sectionNames: string[]): boolean => {
+  return sectionNames.length > 0 && isHeader(sectionNames[0])
+}
+
+// Original requirement was to have a warning for missing header row, leaving this for now
+enum InvalidationType {
+  Error,
+  Warning
+}
+
+interface SectionsSchemaInvalidation {
+  error: string
+  type: InvalidationType
+}
+
+// For validating schema level problems
+interface SectionsSchemaValidator {
+  validate: (sectionName: string[]) => SectionsSchemaInvalidation[]
+}
+
+class SectionNameHeaderValidator implements SectionsSchemaValidator {
+  validate = (sectionNames: string[]): SectionsSchemaInvalidation[] => {
+    const invalidations: SectionsSchemaInvalidation[] = []
+
+    if (!hasHeader(sectionNames)) {
+      invalidations.push({ error: 'Missing SECTION_NAME Header Row', type: InvalidationType.Error })
+    }
+    if (sectionNames.length === 0) {
+      invalidations.push({ error: 'No data', type: InvalidationType.Error })
+    }
+
+    if (sectionNames.length === 1 && isHeader(sectionNames[0])) {
+      invalidations.push({ error: 'No data', type: InvalidationType.Error })
+    }
+    return invalidations
+  }
+}
+
+// For validating row level issues
+interface SectionsRowInvalidation {
+  message: string
+  rowNumber: number
+  type: InvalidationType
+}
+
+interface SectionRowsValidator {
+  validate: (sectionName: string[]) => SectionsRowInvalidation[]
+}
+
+class DuplicateSectionInFileSectionRowsValidator implements SectionRowsValidator {
```

Comment:
You have used the Class based style with react, and infant you are mixing the class and function based style. Nobody said you cannot do it. But I am curious to know what was the reason for this approach?  
I understand that you used the same approach with Canvas Gradebook convertor. I did not ask this question that time but was always there.

Reply:
These aren't React class components (which use `this.state`, etc.), they're just classes. I think it's fine to use classes here if Chris prefers that syntax/programming paradigm. I would probably move the definition out of the React component, as I've suggested elsewhere.

Reply:
I'm pretty sure the validators are not "react" code, they are just typescript in a file that contains a functional react component.  Is there a reason that would be a bad idea?  I still like to use OO design when I can.

Reply:
I have seem people in JS world like Functional style better. I think classes in JS came much later so Functional style is more natural in JS developers. I am not opposed either way but Curious of your choice.

Reply:
I just googled "when did javascript get classes" and it said 2015 so its not new by any means.  I feel like it should be safe to use it. 
Is there 1:1 functional equivalent of classes that javascript programmers prefer?  Is it just that javascript programmers my not understand classes?  I'm still new to javascript and I've been doing  OO programming since college so it just seems  right to me :)

Reply:
To be OO programming dates back to I don't know when. In that comparison I feel Javascript adopt to classes is pretty Late (2015). This is just good conversation on Language.  

`Is there 1:1 functional equivalent of classes that javascript programmers prefer? ` I don't know even I am learning this.